util/src/org/commcare/util/cli/EntityScreen.java

```diff
@@ -0,0 +1,231 @@
+        
+        mHeader = this.createHeader(shortDetail);
+        
+        Vector<TreeReference> references = inflateReference(needed.getNodeset());
+        
+        rows = new String[references.size()];
+        
+        int i = 0;
+        for(TreeReference entity : references) {
+            rows[i] = createRow(entity, shortDetail);
+            ++i;
+        }
+        
+        
+        this.mChoices = new TreeReference[references.size()];
+        references.copyInto(mChoices);
+        setTitle(mTitle);
+    }
+    
+    private String createRow(TreeReference entity, Detail shortDetail) {
+        EvaluationContext context = new EvaluationContext(mSession.getEvaluationContext(), entity);
+        
+        DetailField[] fields = shortDetail.getFields();
+        
+        StringBuilder row = new StringBuilder();
+        int i = 0;
+        for(DetailField field : fields) {
+            Object o = field.getTemplate().evaluate(context);
+            String s;
+            if(!(o instanceof String)) {
+                s = "";
+            } else {
+                s = (String)o;
+            }
+            
+            int widthHint = SCREEN_WIDTH / fields.length;
+            try {
+                widthHint = Integer.parseInt(field.getTemplateWidthHint());
+            } catch(Exception e) {
+                //Really don't care if it didn't work
+            }
+            addPaddedStringToBuilder(row, s, widthHint);
+            i++;
+            if(i != fields.length) {
+                row.append(" | ");
+            }
+        }
+        return row.toString();
+    }
+    
+    //So annoying how identical this is...
+    private String createHeader(Detail shortDetail) {
+        EvaluationContext context = mSession.getEvaluationContext();
+        
+        DetailField[] fields = shortDetail.getFields();
+        
+        StringBuilder row = new StringBuilder();
+        int i = 0;
+        for(DetailField field : fields) {
+            Object o = field.getHeader().evaluate(context);
+            String s;
+            if(!(o instanceof String)) {
+                s = "";
+            } else {
+                s = (String)o;
+            }
+            
+            int widthHint = SCREEN_WIDTH / fields.length;
+            try {
+                widthHint = Integer.parseInt(field.getHeaderWidthHint());
+            } catch(Exception e) {
+                //Really don't care if it didn't work
+            }
+            addPaddedStringToBuilder(row, s, widthHint);
+            i++;
+            if(i != fields.length) {
+                row.append(" | ");
+            }
+        }
+        return row.toString();
+    }
+    
+    
+    private void addPaddedStringToBuilder(StringBuilder builder, String s, int width) {
+        if(s.length() > width) {
+            builder.append(s, 0, width);
+            return;
+        } 
+        builder.append(s);
+        if(s.length() == width){
+            //nothing
```

Comment:
Try to avoid conditionals with empty bodies. Often I see them in the code with TODO code. I think it is cleaner to not include the empty case, but rather leave a comment above/below the conditional stating how the contrary might be handled.